ocaml/database/master_connection.ml

```diff
@@ -215,6 +221,23 @@ let do_db_xml_rpc_persistent_with_reopen ~host:_ ~path (req : string) :
     else if !backoff_delay > 256.0 then
       backoff_delay := 256.0
   in
+  let rec delay = function
+    | d when d > 16.0 -> (
+        (* Sleep for a small time in case that the coordinator can connect but can't accept *)
+        delay (d /. 2.0) ;
+        match Atomic.get in_call_from_coordinator with
+        | true ->
+            debug
+              "%s: The coordinator is calling me. Wake up and retry connecting."
+              __FUNCTION__ ;
+            Atomic.set in_call_from_coordinator false
+        | false ->
+            delay (d /. 2.0)
```

Comment:
It's better if the read and write operations happen atomically:
```suggestion
        let changed = Atomic.compare_and_set in_call_from_coordinator true false in
        if changed then
          debug
            "%s: The coordinator is calling me. Wake up and retry connecting."
            __FUNCTION__ 
        else
          delay (d /. 2.0)
```
https://v2.ocaml.org/api/Atomic.html#VALcompare_and_set